docs/proof.md

```diff
@@ -0,0 +1,555 @@
+
+Changes to the original extension include being able to click on the proof label (e.g. "Theorem 1"), and having a link to that proof anchor. We have also updated the styles from both Sphinx and JupyterBook to be more distinct from admonitions.
+
+You can also reference proofs with any cross-reference syntax (including the `{prf:ref}` role). We recommend the markdown link syntax.
+:::
+
+```{list-table} Proof kinds that can be used as directives
+:name: proof-list
+:header-rows: 0
+
+* - `prf:algorithm`
+  - `prf:axiom`
+  - `prf:conjecture`
+* - `prf:corollary`
+  - `prf:criteria`
+  - `prf:definition`
+* - `prf:example`
+  - `prf:lemma`
+  - `prf:observation`
+* - `prf:property`
+  - `prf:proposition`
+  - `prf:proof`
+* - `prf:remark`
+  - `prf:theorem`
+  -
+```
+
+The following options for proof directives are supported:
+
+- `label`: text
+
+  A unique identifier for your theorem that you can use to reference it with `{prf:ref}`. Cannot contain spaces or special characters.
+
+- `class`: text
+
+  Value of the theorem’s class attribute which can be used to add custom CSS or JavaScript. This can also be the optional `dropdown` class to initially hide the proof.
+
+- `nonumber`: flag (empty)
+
+  Turns off auto numbering.
+
+## Referencing Proofs
+
+You can refer to a proof using the standard link syntax:
+
+- `[](#my-theorem)`, creates [](#my-theorem)
+- `[{name}](#my-theorem)` creates [{name}](#my-theorem)
+- `[{number}](#my-theorem)` creates [{number}](#my-theorem)
+- `[See Theorem](#my-theorem)` creates [See Theorem](#my-theorem)
+
+:::{tip} Compatibility with Sphinx Proof
+:class: dropdown
+
+You may also use the the `{prf:ref}` role like: `` {prf:ref}`my-theorem` ``, which will replace the reference with the theorem number like so: {prf:ref}`my-theorem`. When an explicit text is provided, this caption will serve as the title of the reference. For example, `` {prf:ref}`Orthogonal-Projection-Theorem <my-theorem>`  `` will produce: {prf:ref}`Orthogonal-Projection-Theorem <my-theorem>`.
+:::
+
+## Hiding Proof Content
+
+To hide the directive, simply add `:class: dropdown` as a directive option.
+
+**Example**
+
+```{prf:theorem}
+:class: dropdown
+
+This is an example of how to hide the content of a directive.
+```
+
+**MyST Syntax**:
+
+````md
+```{prf:theorem}
+:class: dropdown
+
+This is an example of how to hide the content of a directive.
+```
+````
+
+## Proof Examples
+
+### Proofs
+
+:::{prf:proof}
+:label: full-proof
+We'll omit the full proof.
+
+But we will prove sufficiency of the asserted conditions.
+
+To this end, let $y \in \mathbb R^n$ and let $S$ be a linear subspace of $\mathbb R^n$.
+
+Let $\hat y$ be a vector in $\mathbb R^n$ such that $\hat y \in S$ and $y - \hat y \perp S$.
```

Comment:
@mmcky I have copied the docs over here to have something to work with as examples. If there is any issue with that, please let me know and we can come up with other examples. I did leave a link to the original work noting that it is copied over.